### PR TITLE
Add confirmation checkboxes to release deletion

### PR DESCRIPTION
**Pull request by mm**
Addresses #13876 by adding in additional checkboxes a user has to go through to confirm the intent of their actions when trying to delete a release before clicking The Big Red Button ™️ 

The goal is to make this similar to the process a user has to go through when deleting a project, where the user has to physically click on each checkbox to enable the display of the deletion button.

Lil video demonstration:

https://github.com/pypi/warehouse/assets/639901/287b752f-6c8f-47f0-a8c5-a0fab520bcf5

### CODE REVIEW

File: warehouse/templates/manage/project/release.html [hunk 1]
```diff
@@ -192,8 +192,13 @@ <h3>{% trans %}Delete release{% endtrans %}</h3>
       <p>{% trans %}You will not be able to re-upload a new distribution of the same type with the same version number.{% endtrans %}</p>
       <p>{% trans %}Deletion will break any downstream projects relying on a pinned version of this package. It is intended as a last resort to address legal issues or remove harmful releases.{% endtrans %}</p>
       <p>{% trans yank_href=request.help_url(_anchor='yanked'), post_href='https://www.python.org/dev/peps/pep-0440/#post-releases', title=gettext('External link') %}Consider <a href="{{ yank_href }}" title="{{ title }}" target="_blank" rel="noopener">yanking</a> this release, making a new release or a <a href="{{ post_href }}" title="{{ title }}" target="_blank" rel="noopener">post release</a> instead.{% endtrans %}</p>
+      <ul class="no-bullets">
```
Comment (ewdurbin) [inline warehouse/templates/manage/project/release.html:195]: Suggestions for the acknowledgements here to make them relevant to the action and consistent with the project delete confirmations:

- [ ] I understand that I am permanently deleting all distributions for the {{ release.version }} release of this project.
- [ ] I understand that my users will no longer be able to install this release of this project.
- [ ] I understand that I will <a href="https://pypi.org/help/#file-name-reuse">not be able to re-upload any deleted distribution files using the same file names.</a>.
- [ ] I understand that I will not be able to undo this.
- [ ] I understand that the PyPI administrators will not be able to undo this.

**Review by ewdurbin — COMMENTED** [review]
Thank you for your contribution @mm! One request to make things a little more consistent with the project delete variant: https://github.com/pypi/warehouse/blob/d9c18a0c62cbfcd5fd959e35a4fe8a445010dbca/warehouse/templates/manage/project/settings.html#L275-L283

**Comment by ewdurbin** [comment]
Feel welcome to say "No!" but I wanted to ask if while you're in here, you might consider adding similar warnings to the [file deletion](https://github.com/pypi/warehouse/blob/d9c18a0c62cbfcd5fd959e35a4fe8a445010dbca/warehouse/templates/manage/project/release.html#L105-L120)

- [ ] I understand that my users will no longer be able to install this file for this release of this project.
- [ ] I understand that I will <a href="https://pypi.org/help/#file-name-reuse">not be able to re-upload a using the same file name</a> {{ file.filename }}.
- [ ] I understand that I will not be able to undo this.
- [ ] I understand that the PyPI administrators will not be able to undo this.

File: warehouse/templates/manage/project/release.html (hunk 1 above)
Comment (mm) [reply warehouse/templates/manage/project/release.html:195]: Those sound great! I've added the additional ones in [c6f1259](https://github.com/pypi/warehouse/pull/13930/commits/c6f1259936646d1e03b0f454e66345a5f7705695)

**Comment by mm** [comment]
> Feel welcome to say "No!" but I wanted to ask if while you're in here, you might consider adding similar warnings to the [file deletion](https://github.com/pypi/warehouse/blob/d9c18a0c62cbfcd5fd959e35a4fe8a445010dbca/warehouse/templates/manage/project/release.html#L105-L120)

for sure! Can add it as part of this PR 😄

**Comment by ewdurbin** [comment]
> > Feel welcome to say "No!" but I wanted to ask if while you're in here, you might consider adding similar warnings to the [file deletion](https://github.com/pypi/warehouse/blob/d9c18a0c62cbfcd5fd959e35a4fe8a445010dbca/warehouse/templates/manage/project/release.html#L105-L120)
> 
> for sure! Can add it as part of this PR 😄 As that opens a modal, do you think it'd be valuable as a similar checkbox/"all must be selected to proceed" style, or would it be better to itemize them instead because there's already a version number confirmation in that modal?

I think we should do the same checkbox gauntlet! Thank you for being open to this.

**Comment by ewdurbin** [comment]
@mm I'm going to merge this and file a new issue for the individual file delete modal!

**Review by ewdurbin — APPROVED** [review]

**Comment by mm** [comment]
Sounds great! 🔥 I have a rough version of this working locally, will add more detail in the new issue

**Pull request merged**